src/test/java/com/fasterxml/jackson/module/jsonSchema/TestGenerateJsonSchema.java

```diff
@@ -280,14 +282,23 @@ public void testSinglePropertyDependency() throws Exception {
         assertNotNull(result);
 
         String schemaString = MAPPER.writeValueAsString(jsonSchema);
```

Comment:
Your modification looks great! There may be three ways to improve it.
1. Is it possible to modify ```writeValueAsString``` so that the order of ```dependencies``` and properties is deterministic as well?
2. The code can be less redundant with a string of properties and a string of dependencies. Two different orders can be represented by the two strings. Now, there are many repeated lines of code in ```possibleString1``` and ```possibleString2```.
3. Naming of ```possibleString1``` and ```possibleString2``` does not show the semantics of the variables. Maybe it would be better to use some name like ```possibleSchemaString1``` and ```possibleSchemaString2```.

Reply:
Thank you Yan,

Based on your suggestion 1, I have found I way to sort the string so that dependencies will always come before id. I will open the PR to the original repo now.

Mingqing

Reply:
No problem. You can just call me Yifei.